### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: js/src/run_trees.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export class RunTree implements BaseRun {
       api_key: getEnvironmentVariable("LANGCHAIN_API_KEY"),
       caller_options: {},
       start_time: Date.now(),
-      end_time: Date.now(),
+      end_time: null,
```
Comment (dqbd) [inline js/src/run_trees.ts:65]: _nit_: shouldn't that be `undefined` instead? Or maybe we should change `end_time` data type?

**Review by dqbd — COMMENTED** [review]
Smaller nitpicks :)

**Review by dqbd — APPROVED** [review]
Smaller nitpicks :)

File: js/src/tests/client.int.test.ts [hunk 2]
```diff
@@ -264,8 +279,11 @@ test("test create dataset", async () => {
     apiUrl: "http://localhost:1984",
   });
   const datasetName = "__test_create_dataset";
-  const datasets = await langchainClient.listDatasets({});
-  if (datasets.map((dataset) => dataset.name).includes(datasetName)) {
+  const datasets = [];
+  for await (const dataset of langchainClient.listDatasets({})) {
+    datasets.push(dataset);
+  }
+  if (datasets.some((dataset: any) => dataset.name === datasetName)) {
```
Comment (dqbd) [inline js/src/tests/client.int.test.ts:286]: _nit_: Using `toArray` we would be able to get rid of `any`

File: js/src/client.ts [hunk 3]
```diff
@@ -108,6 +108,14 @@ const raiseForStatus = async (response: Response, operation: string) => {
   }
 };
 
+export async function toArray<T>(iterable: AsyncIterable<T>): Promise<T[]> {
```
[the remark this answers is not in the document]
Comment (hinthornw) [reply js/src/client.ts:111]: Hm probably not to start ya

File: js/src/run_trees.ts (hunk 1 above)
Comment (hinthornw) [reply js/src/run_trees.ts:65]: Yes!

**Pull request merged**